app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -375,19 +375,22 @@ private void displayCurrentInfo() {
         df.setMaximumFractionDigits(0);
 
         boolean isDexbridge = CollectionServiceStarter.isDexbridgeWixel(getApplicationContext());
+        boolean displayBattery = prefs.getBoolean("display_bridge_battery",false);
         int bridgeBattery = prefs.getInt("bridge_battery", 0);
 
-        if (isDexbridge) {
+        if (isDexbridge && displayBattery) {
             if (bridgeBattery == 0) {
-                dexbridgeBattery.setText("Waiting for packet");
+                dexbridgeBattery.setText("xBridge Battery: Unknown, Waiting for packet");
+                dexbridgeBattery.setTextColor(Color.WHITE);
             } else {
-                dexbridgeBattery.setText("Bridge Battery: " + bridgeBattery + "%");
+                dexbridgeBattery.setText("xBridge Battery: " + bridgeBattery + "%");
             }
-            if (bridgeBattery < 50) dexbridgeBattery.setTextColor(Color.YELLOW);
-            if (bridgeBattery < 25) dexbridgeBattery.setTextColor(Color.RED);
+            if (bridgeBattery < 50 && bridgeBattery >25) dexbridgeBattery.setTextColor(Color.YELLOW);
+            if (bridgeBattery <= 25) dexbridgeBattery.setTextColor(Color.RED);
             else dexbridgeBattery.setTextColor(Color.GREEN);
```

Comment:
You are right, but I had not noticed that before.  Fixed the logic now I
believe.
I cannot see a reason why you would always have the red Waiting for Packet
message showing permanently though.  I will try and run it up on my test
phone and see if I can reproduce.
Cheers

On Fri, Dec 11, 2015 at 9:27 AM, Johan Degraeve <notifications@github.com>
wrote:

> In app/src/main/java/com/eveningoutpost/dexdrip/Home.java
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/209#discussion_r47296192>
> :
>
> >              }
> > -            if (bridgeBattery < 50) dexbridgeBattery.setTextColor(Color.YELLOW);
> > -            if (bridgeBattery < 25) dexbridgeBattery.setTextColor(Color.RED);
> > +            if (bridgeBattery < 50 && bridgeBattery >25) dexbridgeBattery.setTextColor(Color.YELLOW);
> > +            if (bridgeBattery <= 25) dexbridgeBattery.setTextColor(Color.RED);
> >              else dexbridgeBattery.setTextColor(Color.GREEN);
>
> note that it wil still be reset to green whenver > 25, it will never be
> yellow
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/209/files#r47296192>
> .
>



-- 
John Stevens
"You are how you live, not what you have."